homeassistant/components/ozw/services.py

```diff
@@ -37,6 +43,99 @@ def async_register(self):
             ),
         )
 
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_SET_CONFIG_PARAMETER,
```

Comment:
I think that these need to be bound to an entity too. 

Also, why would we allow this to be done in services? Are these things we want to automate?

Reply:
Some devices have config parameters that are useful to automate. For example Inovelli dimmers have config params for changing the color/brightness of the LED on the switch, useful for notifications, or adjusting the default brightness when the physical switch is triggered

Reply:
Yes these would be to automate for instance "If garage door left open after 7pm turn LED on Inovelli light switch to fast blinking red", Homeseer switches have similar config parameters to automate LEDs for status.

Example of one of my automations from the old zwave integration:
```
# Alarm Indicators on switches
- alias: Alarm Indicator Lights (armed_away)
  trigger:
    platform: state
    entity_id: alarm_control_panel.house_alarm
    to: "armed_away"

  condition:
    condition: time
    after: "20:00:00"
    before: "05:00:00"

  action:
    - service: zwave.set_config_parameter
      data:
        node_id: 26
        parameter: 8
        value: 50268673
```
This would make the LED fast blink red to indicate the system is armed.

Reply:
I can definitely get them bound to an entity, would need to add it to multiple areas like switch, light, and sensor.
These are the device types I can recall off the top of my head that use parameters to change LED status lights.

Reply:
If we keep them general purpose we can't bind the services to an entity. We would need to make them more specific to bind them to an entity.

Reply:
Correct, the set config paramater is bound to a device (Z-Wave node) and not to an entity.

BTW: I did not yet migrate this service over from the old (and temp) implementation because I was thinking how to make this more userfriendly. For example now the parameter is the index of the value. Some improvements from the top of my mind:

1) Allow multiple ways to set a config value. I'm thinking of:
  - parameter_index: set value by the index of the configuration value/setting
  - parameter: set value by label (more userfriendly to read in automations)

2) Allow setting list values by both int value and text label (more friendly to read/understand in automations)

Reply:
So I was thinking, what if we make it an entity service that can take ANY OZW entity and will set the parameter on the correct OZW Device? I want to try to avoid having the user have to know the internal OZW identifier and instead be able to use the entity IDs of the OZW entities.

Reply:
> So I was thinking, what if we make it an entity service that can take ANY OZW entity and will set the parameter on the correct OZW Device? I want to try to avoid having the user have to know the internal OZW identifier and instead be able to use the entity IDs of the OZW entities.

I like this idea as it's more userfriendly but some users are kind of attached to the node id's as that's what they're dealing with in the Z-Wave world (such as OZW Admin). Can we have both ? So you either supply a valid entity_id OR node id ?

Reply:
Yeah I guess that makes sense, especially for things that don't have entity IDs (remotes or something).

Reply:
So, in the spirit of bite sized PRs, breaking down the entity services into seperate PRs would be the thing to do here?

Reply:
Yes, we can add entity support in a next PR 👍

Reply:
Yeah, we can add some logic to call these services by supplying an entity_id later. Great addition!

Reply:
I have one more up my sleeve :P